realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -773,16 +775,123 @@ private Scanner getFullStringScanner(InputStream in) {
         if (objects == null) {
             return new ArrayList<E>();
         }
-
+        Map<RealmModel, RealmObjectProxy> cache = new HashMap<RealmModel, RealmObjectProxy>();
         ArrayList<E> realmObjects = new ArrayList<E>();
         for (E object : objects) {
-            realmObjects.add(copyToRealm(object));
+            checkNotNullObject(object);
+            realmObjects.add(copyOrUpdate(object, false, cache));
         }
 
         return realmObjects;
     }
 
     /**
+     * Insert a list of an unmanaged RealmObjects. This is generally faster than {@link #copyToRealm(Iterable)} since it
+     * doesn't return the inserted elements, and performs minimum allocations and checks.
+     * After being inserted any changes to the original objects will not be persisted.
+     * <p>
+     * Please note:
+     * <ul>
+     * <li>We don't check if the provided objects are already managed or not, so inserting a managed object will duplicate it</li>
+     * <li>We don't create (nor return) a managed {@link RealmObject} for each element</li>
+     * <li>Copying an object will copy all field values. Any unset field in the object and child objects will be set to their default value if not provided</li>
+     * </ul>
+     * <p>
+     * If you want the previous checks and the returned managed {@link RealmObject} use {@link #copyToRealm(Iterable)}, otherwise if
```

Comment:
remove "previous"
"the returned managed {@link RealmObject}" -> "the managed {@link RealmObject} returned, "